crates/bevy_ecs/src/world/mod.rs

```diff
@@ -331,13 +334,16 @@ impl World {
                 .iter()
                 .enumerate()
                 .map(|(archetype_row, archetype_entity)| {
+                    let entity = archetype_entity.entity();
                     let location = EntityLocation {
                         archetype_id: archetype.id(),
                         archetype_row: ArchetypeRow::new(archetype_row),
                         table_id: archetype.table_id(),
                         table_row: archetype_entity.table_row(),
                     };
-                    EntityRef::new(self, archetype_entity.entity(), location)
+
+                    // SAFETY: entity exists and location is validly constructed above
```

Comment:
Can you be a bit more specific about how we know that the location is valid? Something like "location accurately specifies the archetype where the entity is stored" would be more clear IMO.

Reply:
Oops, this suggestion was slightly inaccurate/incomplete. It's not just the archetype that matters -- a better way of phrasing it could be "location accurately specifies where the entity is stored".

Not sure how important this is, though.

Reply:
I think this basically covers it: this isn't a particularly tricky bit of unsafe. Feel free to make a follow-up PR if you think it's worthwhile though :)